b42_prepush_check.sh

```diff
@@ -29,27 +34,30 @@ summary_status="${summary_status}.${uncommited_status}"
 
 echo "== Filename to function name"
 not_found=0
+OLDIFS="${IFS}"
+IFS=$'\n'
 for fname in $(find . -name 'ft_*.c'); do
-	func_name=$(basename -s .c $fname);
+	func_name="$(basename -- "$fname" .c)"
 	# grep "^/.*$func_name" "$fname"	# Check for fname in header
-	egrep -H "^(char|int|void)\s+\*?$func_name\(" $fname # Check for the function declaration
+	grep -EH "^(char|int|void|long)\s+\*+${func_name}\(" "$fname" # Check for the function declaration
```

Comment:
nice catch, thanks!
I also used to fail that check today with `unsigned int`. We could extend it here as well or in separate PR.

Reply:
one more comment:
I had `\*?` there to catch when function returns pointer.
With your suggestion `\*+` you try to catch return of `pointer to pointer`, right? nice catch, we didn't have that yet.
But that breaks if function returns just value :-(
```
for fname in $(find . -name 'ft_*.c'); do func_name=$(basename -s .c $fname); grep -EH "^(char|int|void|long)\s+\*+${func_name}\(" "$fname"; done
./ex03/ft_strncat.c:char	*ft_strncat(char *dest, char *src, unsigned int nb)
./ex02/ft_strcat.c:char	*ft_strcat(char *dest, char *src)
```
vs
```
for fname in $(find . -name 'ft_*.c'); do func_name=$(basename -s .c $fname); grep -EH "^(char|int|void|long)\s+\**${func_name}\(" "$fname"; done
./ex00/ft_strcmp.c:int	ft_strcmp(char *s1, char *s2)
./ex01/ft_strncmp.c:int	ft_strncmp(char *s1, char *s2, unsigned int n)
./ex03/ft_strncat.c:char	*ft_strncat(char *dest, char *src, unsigned int nb)
./ex02/ft_strcat.c:char	*ft_strcat(char *dest, char *src)
```
I will make a suggestion there